### ISSUE TITLE
different speed on Ubuntu and Windows10

### ISSUE THREAD

**Issue by JingtaoWang22** [rstudio/reticulate#845]
Hi, 

I have some python code for running sklearn's PCA. And I need to call it from R.
On my laptop (windows), python and R have same speed.
However on my PC (ubuntu), python (2 seconds) is significantly faster than R (4 minutes). 
So does reticulate treat different operating systems differently? How can I improve the speed in Ubuntu when using R?

Thank you!

python:
`class Dim_TruncatedSVD(Unit):
    def __init__(self, n_components='knee', n_components_max=30, **kwargs):
        self.n_components = n_components
        self.n_components_max = n_components_max
        self.kwargs = kwargs
    def get(self, x, return_evr=False):
        if self.n_components == 'knee':
            n_components = min(self.n_components_max, np.min(x.shape))
            obj = TruncatedSVD(n_components=n_components, **self.kwargs).fit(x)

            # Construct axis for KneeLocator
            x_axis = list(range(1, n_components + 1))
            y_axis = obj.explained_variance_ratio_
            # Find knee
            n_components = KneeLocator(x_axis, y_axis,
                                       curve='convex',  # Approximately
                                       direction='decreasing'
                                       ).knee
            n_components = max(n_components, 10)
            x_emb = TruncatedSVD(n_components=n_components,
                                 **self.kwargs).fit_transform(x)
        else:
            x_emb = TruncatedSVD(n_components=self.n_components,
                                 **self.kwargs).fit_transform(x)
        if return_evr == True:
            return x_emb, y_axis
        else:
            return x_emb

adata=anndata.read_h5ad(filepath)
x_emb=Dim_TruncatedSVD().get(adata.X)
`

R:
`source_python('py_test.py')`

**Comment by JingtaoWang22** [comment]
I found the reason. It seems that on ubuntu reticulate doesn't support multi-cpu calculation.

**Comment by dkincaid** [comment]
We just ran into this problem too. Why doesn't reticulate on Ubuntu support multi-cpu calculation and how can this be fixed?

**Comment by JingtaoWang22** [comment]
> We just ran into this problem too. Why doesn't reticulate on Ubuntu support multi-cpu calculation and how can this be fixed?

Actually later we fixed this by re-install all of the packages (python and R) we use. We are still not sure about the reason but this might be caused by some old package (so probably not the operating system). Probably numpy? Sorry about not being able to give an explicit solution, but maybe you can try to update some packages.

**Comment by dkincaid** [comment]
Interesting. Thank you for the reply. We'll see if we can figure out which package is causing the issue.
